extra/src/main/java/aima/extra/probability/RandVar.java

```diff
@@ -70,4 +71,17 @@ public boolean equals(Object o) {
 		boolean result = this.name.equals(other.name) && this.domain.equals(other.domain);
 		return result;
 	}
+
+	/**
+	 * Two RandVar objects have the same hashcode if they have the same names.
+	 */
+	@Override
+	public int hashCode() {
+		return Objects.hash(name);
```

Comment:
Note: Objects.hash is intended to be used with a sequence of objects. For a single object as in this case just use name.hashCode